### PR TITLE
FOGL-225 build.sh and Makefile refactoring

### PR DESCRIPTION
**Pull request by Terris-Dianomic**
Add to build.sh:

--install-all-dep
--install-dev-dep

For installing dependencies for testing into virtual environment.

Rename requirements_*.txt to requirements-*.txt per file naming conventions.

Refactoring and error handling.

### CODE REVIEW

**Comment by Terris-Dianomic** [comment]
I'm working more on this for JIRAs 179, 229, and 230

**Comment by Terris-Dianomic** [comment]
Ready for review.

File: src/python/Makefile [hunk 1]
```diff
@@ -36,57 +53,57 @@ clean-build:
...
 	rm -fr .cache
 	rm -fr .tox
 
-install-py-requirements: check-root
-	@echo "-- Installing Python dependencies"
+install-requirements: check-root
 	pip install -r requirements.txt
 
+install-all-requirements: check-root
+	pip install -r requirements-all.txt
```
Comment (praveen-garg) [inline src/python/Makefile]: dev should be all itself. I recommend to keep only -dev i.e. all

File: src/python/Makefile [hunk 2]
```diff
@@ -36,57 +53,57 @@ clean-build:
...
-	pip install -r requirements_dev.txt
+	pip install -r requirements-dev.txt
 
 install-doc-requirements: check-root
-	@echo "-- Installing Sphinx doc dependencies"
-	pip install -r requirements_doc.txt
+	pip install -r requirements-doc.txt
+
+install-lint-requirements: check-root
+	pip install -r requirements-lint.txt
```
Comment (praveen-garg) [inline src/python/Makefile]: lint should be in test itself.

File: src/python/Makefile [hunk 3]
```diff
@@ -36,57 +53,57 @@ clean-build:
...
 
 install-test-requirements: check-root
-	@echo "-- Installing test dependencies"
-	pip install -r requirements_test.txt
+	pip install -r requirements-test.txt
 
+install: install-requirements
+	pip install -e .
+
```
Comment (praveen-garg) [inline src/python/Makefile]: let's add 1 more?

`develop: install-dev-requirements`
------`pip install -e .`

File: src/python/Makefile [hunk 4]
```diff
@@ -36,57 +53,57 @@ clean-build:
...
-	@echo "-- Creating environment file"
 	[ -f foglamp/foglamp-env.yaml ] && echo "foglamp-env already exists" || cp foglamp/foglamp-env.example.yaml foglamp/foglamp-env.yaml
-
+	
 test: py-test doc-build-test
 
-py-test:
-	@echo "-- Executing python tests"
-	pip install tox
+py-test: install-test-requirements
```
Comment (praveen-garg) [inline src/python/Makefile]: we just need tox; tox will use its own env to run; and install dependencies

File: src/python/Makefile [hunk 5]
```diff
@@ -36,57 +53,57 @@ clean-build:
...
 	tox -e py35
 
-doc-build-test:
-	@echo "-- Executing sphinx doc builder tests"
-	pip install tox
+doc-build-test: install-doc-requirements
```
Comment (praveen-garg) [inline src/python/Makefile]: we just need tox; tox will use its own env to run; and install dependencies

File: src/python/Makefile [hunk 6]
```diff
@@ -36,57 +53,57 @@ clean-build:
...
 
-lint-python: install-test-requirements
-	@echo "-- Linting Python files"
+lint-python: install-lint-requirements
 	rm -f pylint*.log
 	pylint $(shell find . -iname "*.py" ! -path './venv*' ! -path './.tox*') > pylint_$(shell date +%FT%T%Z).log || exit 0
 
 lint-js:
-    @echo "-- TODO: add js lint | added here for example, to add later"
+  @echo "-- TODO: add js lint | added here for example, to add later"
```
Comment (praveen-garg) [inline src/python/Makefile]: not used; but Make use tab spacing

File: src/python/build.sh [hunk 7]
```diff
@@ -138,7 +159,7 @@ setup_and_run() {
 
             if [ $? -gt 0 ]
             then
-                pip install virtualenv
+                pip install -r requirements-virtualenv.txt
```
Comment (praveen-garg) [inline src/python/build.sh]: what else we are expecting in requirements-virtualenv.txt; if nothing.. I believe we should just avoid it.

File: src/python/build.sh [hunk 8]
```diff
@@ -170,18 +191,18 @@ setup_and_run() {
...
-            echo "-- Creating virtualenv for ${python_path}"
+            echo "-- Creating virtual environment for ${python_path}"
             virtualenv "--python=$python_path" "$VENV_PATH"
         fi
 
-        echo "-- Activating the virtualenv at $VENV_PATH"
+        echo "-- Activating the virtual environment at $VENV_PATH"
         source "$VENV_PATH/bin/activate"
 
         IN_VENV=$(python -c 'import sys; print ("1" if hasattr(sys, "real_prefix") else "0")')
```
Comment (praveen-garg) [inline src/python/build.sh]: python3

File: src/python/Makefile (hunk 1 above)
Comment (Terris-Dianomic) [inline src/python/Makefile]: You want dev to have requirements-doc.txt?

**Comment by Terris-Dianomic** [comment]
Please review

**Pull request merged**